packages/headless/src/app/commerce-engine/commerce-engine-configuration.ts

```diff
@@ -10,11 +12,17 @@ import {
  *
  * @internal WORK IN PROGRESS. DO NOT USE IN ACTUAL IMPLEMENTATIONS.
  */
-export interface CommerceEngineConfiguration extends EngineConfiguration {}
+export interface CommerceEngineConfiguration extends EngineConfiguration {
+  context: ContextOptions;
+}
 
 export const commerceEngineConfigurationSchema =
   new Schema<CommerceEngineConfiguration>({
     ...engineConfigurationDefinitions,
+    context: new RecordValue({
```

Comment:
I wonder, should we also allow setting the state of the cart through the engine's configuration?

I think this would work well in multi page apps: one could keep the current state of the cart in local storage and restore it on each page load as they re-initialize the commerce engine.

Reply:
Hmmm, interesting idea. I thought we could use `preloadedState` for that, but that property requires the entire engine state. I'd propose we revisit this in https://coveord.atlassian.net/browse/CAPI-774